src/RootContainer.js

```diff
@@ -1,21 +1,48 @@
 import React, { useState, useEffect } from 'react';
 import { queryData } from './query';
 import GeneOntologyNetwork from './components/GeneOntologyNetwork';
+import Controls from './components/Controls';
+import Loading from './components/Loading';
 
 const RootContainer = ({ serviceUrl }) => {
 	const [data, setData] = useState([]);
+	const [selectedOntology, changeOntology] = useState('biological_process');
+	const [loading, setLoading] = useState(true);
 	useEffect(() => {
+		setLoading(true);
 		queryData({
 			serviceUrl: serviceUrl,
-			geneId: '1205472,128,2314'
+			geneId: '1205472,128,2314',
+			ontology: selectedOntology
 		}).then(data => {
 			setData(data);
+			setLoading(false);
+			if (data.length) document.getElementById(selectedOntology).checked = true;
 		});
-	}, []);
+	}, selectedOntology);
+
+	function updateFilters(ev) {
+		document.getElementById(selectedOntology).checked = false;
```

Comment:
You shouldn't need to set the radio button with a DOM manipulation. Instead pass `selectedOntology` as a prop into the `Controls` component and add the attribute `checked={selectedOntology === term}`. (See https://reactjs.org/docs/forms.html for more tips on using inputs with React.)

You should also change the similar DOM manipulation in the queryData promise above to call `changeOntology` instead.

Reply:
Yes, the things you mentioned also came in my mind.
I also thought not to hardcode the ontology namespaces but in `go-term-viz` the namespaces were hardcoded, so I thought we might need only those 3 of them.
For avoiding the re-run of query and saving the state once, I think filtering is better on DB instead of doing it on frontend since it won't much much sense and can give a bad UX. 
What do you think?

Reply:
Aah! That's so silly of me. Thanks for letting me know about this issue.